katpoint/test/test_projection.py

```diff
@@ -248,10 +347,49 @@ def test_corner_cases(self):
         ae = np.array(self.plane_to_sphere(0.0,  -np.pi / 2.0, 0.0, -1.0))
         assert_angles_almost_equal(ae, [np.pi, -np.pi / 4.0], decimal=12)
 
+    def test_out_of_range_cases(self):
+        """TAN projection: test out-of-range cases."""
+        # SPHERE TO PLANE
+        # Points outside allowed domain on sphere
+        with OutOfRange.set_treatment('raise'):
+            self.assertRaises(OutOfRangeError,
+                              self.sphere_to_plane, 0.0, np.pi, 0.0, 0.0)
+            self.assertRaises(OutOfRangeError,
+                              self.sphere_to_plane, 0.0, 0.0, np.pi, 0.0)
+            self.assertRaises(OutOfRangeError,
+                              self.sphere_to_plane, 0.0, 0.0, 0.0, np.pi)
+        with OutOfRange.set_treatment('nan'):
+            xy = np.array(self.sphere_to_plane(0.0, np.pi, 0.0, 0.0))
+            np.testing.assert_array_equal(xy, [np.nan, np.nan])
+            xy = np.array(self.sphere_to_plane(0.0, 0.0, np.pi, 0.0))
+            np.testing.assert_array_equal(xy, [np.nan, np.nan])
+            xy = np.array(self.sphere_to_plane(0.0, 0.0, 0.0, np.pi))
+            np.testing.assert_array_equal(xy, [np.nan, np.nan])
+        with OutOfRange.set_treatment('clip'):
+            xy = np.array(self.sphere_to_plane(0.0, np.pi, 0.0, 0.0))
+            np.testing.assert_almost_equal(xy, [0.0, -1e6], decimal=4)
+            xy = np.array(self.sphere_to_plane(0.0, 0.0, np.pi, 0.0))
+            np.testing.assert_almost_equal(xy, [-1e6, 0.0], decimal=4)
+            xy = np.array(self.sphere_to_plane(0.0, 0.0, 0.0, np.pi))
+            np.testing.assert_almost_equal(xy, [0.0, 1e6], decimal=4)
```

Comment:
This feels a bit dirty to me, in that you're testing a magic number that can't be determined from the API. Maybe a better test is to convert back to az, el and check that you are close to where you started?

Reply:
I agree that it is a bit dirty but in this case I think I can live with the mud. The test is mostly to detect regressions, and in that case some inside knowledge is OK. I found those numbers by running `self.sphere_to_plane(0.0, 0.0, 0.0, np.pi)` in the first place and recording the output in the test.